tests/migrations.rs

```diff
@@ -114,3 +115,69 @@ fn check_non_null_column_without_default() {
         }
     }
 }
+
+fn get_up_migrations() -> Vec<PathBuf> {
+    let root = env!("CARGO_MANIFEST_DIR");
+    let migrations_dir = PathBuf::from(root).join("migrations");
+
+    let mut migrations = Vec::new();
+    for entry in std::fs::read_dir(migrations_dir).unwrap() {
+        let path = entry.unwrap().path();
+
+        if path.is_file() && path.to_string_lossy().contains(".up.sql") {
+            migrations.push(path);
+        }
+    }
+
+    // Sort by timestamp to enforce migration order
+    migrations.sort_by(|a, b| a.file_name().unwrap().cmp(b.file_name().unwrap()));
+    migrations
+}
+
+fn get_test_data_path(migration_path: &Path) -> PathBuf {
+    let filename = migration_path.file_name().unwrap().to_str().unwrap();
+    // e.g. "20240517094752_create_build.up.sql" -> "20240517094752_create_build"
+    let migration_name = filename.split('.').next().unwrap();
+
+    let root = env!("CARGO_MANIFEST_DIR");
+    PathBuf::from(root)
+        .join("tests/data/migrations")
+        .join(format!("{}.sql", migration_name))
+}
+
+#[test]
+fn check_migrations_have_sample_data() {
+    for migration_path in get_up_migrations() {
+        let test_data_path = get_test_data_path(&migration_path);
+
+        assert!(
+            test_data_path.exists(),
+            "Migration {:?} does not have an associated test data file at {:?}",
+            migration_path,
+            test_data_path
+        );
+    }
+}
+
+#[sqlx::test(migrations = false)]
```

Comment:
Please add a brief comment above this function that explains what the function does.